File: tests/eval_c/540.c
#include <assert.h>
#include <string.h>

void f540(const char *s) {
    int i = 0;
    while (i < strlen(s)) {
        if (s[i] == 'a') {
            assert(s[i + 1] == 'b');
            assert(s[i + 2] == 'b');
            i += 3;
        } else {
            i++;
        }
    }
}

File: benchmarks/oopsla25/subjects/540.py
import sys

def f540(s: str):
  i = 0
  while i < len(s):
    if s[i] == "a":
      assert s[i+1] == "b"
      assert s[i+2] == "b"
      i += 3
    else:
      i += 1

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f540(my_string)
